src/Watchers/Warden.Watchers.ServerStatus/PortAvailabilityWatcher.cs

```diff
@@ -0,0 +1,117 @@
+﻿namespace Warden.Watchers.ServerStatus
+{
+    using System;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Threading.Tasks;
+    using global::Warden.Core;
+
+    public class PortAvailabilityWatcher : IWatcher
+    {
+        /// <summary>
+        /// Default name for this watcher.
+        /// </summary>
+        public const string DefaultName = "Port availability Watcher";
+
+        /// <summary>
+        /// Current name of this watcher.
+        /// </summary>
+        public string Name { get; }
+        
+        /// <summary>
+        /// Actual configuration of current instance of this watcher.
+        /// </summary>
+        public PortAvailabilityConfiguration Configuration { get; }
+
+        /// <summary>
+        /// Ctor of PortAvailabilityWatcher.
+        /// </summary>
+        /// <param name="name">Name of this watcher.</param>
+        /// <param name="configuration">Configuration to use.</param>
+        protected PortAvailabilityWatcher(string name, PortAvailabilityConfiguration configuration)
+        {
+            this.Name = name;
+            this.Configuration = configuration;
+        }
+
+        /// <summary>
+        /// Performs a single check for the watcher.
+        /// </summary>
+        /// <returns>Instance of IWatcherCheckResult.</returns>
+        public async Task<IWatcherCheckResult> ExecuteAsync()
```

Comment:
This method could be refactored into 2 separate methods. For example, the first one would check availability of the _client_ and _dnsResolver_  and the second one would execute an action.